docker-compose.yml

```
@@ -0,0 +1,21 @@
+version: "1"
+
+services:
+  to-do-app:
+    build: 
+      context: .
+      target: development
+    ports:
+      - "5000:5000"
+    env_file:
+      - path: .env  
+    volumes:
+      - ./:/usr/src/app:rw
```

I suspect this broad bind mount is part of your issue with various things throwing errors, I'd lean towards just mapping the code  personally so that you've got less risk of changes in the container polluting your host:
```suggestion
      - ./todo_app:/usr/src/app/todo_app:rw
```
And if you want to keep the `poetry install` stage in the ENTRYPOINTs, then we might also bind in the `pyproject.toml` and `poetry.lock` files explicitly as two extra lines here

---

I kept the mount the same for the debug environment since otherwise it doesn't have my .vscode folder and can't run it with `--host 0.0.0.0` without any extra steps, but that seems to be fine as long as you don't run pytest inside the debug container